index.js

```diff
@@ -339,3 +339,86 @@ module.exports.random = {
     return s * Math.pow(2, (-64 - e))
   }
 }
+
+/**
+ * A dictionary of header values, i.e '{ 'header-name': 'header-value' }'
+ * @typedef {Object.<string, (string | string[] | undefined)>} HeaderLike
+ */
+
+/**
+ * Uses Ed25519: a public-key signature system: {@link https://ed25519.cr.yp.to/} {@link https://datatracker.ietf.org/doc/html/draft-cavage-http-signatures-12}.
+ * Signs the message using the secret key and returns a signature.
+ * @see {nacl.sign.detached}.
+ *
+ * @method
+ * @param {string} keyId - an opaque string that the server can use to look up the component they need to validate the signature.
+ * @param {string} secretKey - hex encoded secret key to sign the message.
+ * @param {HeaderLike} headers - headers containing the properties to sign.
+ * @returns {string}
+ */
+module.exports.ed25519HttpSign = function (keyId /* string */, secretKey /* string */, headers = {} /* dictionary */) {
+  if (!secretKey) throw new Error('secret key is required')
+  if (!keyId) throw new Error('key id is required')
+  if (Object.keys(headers).length === 0) throw new Error('headers are required')
+
+  const headerKeys = []
+  const message = Object.entries(headers)
+    .map(([key, value]) => {
+      headerKeys.push(key)
+      return `${key}: ${value}`
+    }).join('\n')
+
+  // Generate signature using the ed25519 algorithm.
+  const sign = nacl.sign.detached(
+    Uint8Array.from(Buffer.from(message)),
+    Uint8Array.from(Buffer.from(secretKey, 'hex'))
+  )
+
+  const signature = Buffer.from(sign).toString('base64')
+  return `keyId="${keyId}",algorithm="ed25519",headers="${headerKeys.join(' ')}",signature="${signature}"`
+}
+
+/**
+ * Uses Ed25519: a public-key signature system: {@link https://ed25519.cr.yp.to/} {@link https://datatracker.ietf.org/doc/html/draft-cavage-http-signatures-12}.
+ * Verifies the signature for the message and returns parsed fields from the signature.
+ * @see {nacl.sign.detached.verify}.
+ *
+ * @method
+ * @param {string} publicKey - hex encoded public key to verify the signature.
+ * @param {HeaderLike} headers - headers containing the signature for verification.
+ * @returns {{ algorithm: string, headers: string[], keyId: string, signature: string, verified: boolean }}.
```

Comment:
i think the period at the end of this line is causing some issues (jsdoc thinks it's a comment that just consists of `.`)